### PR TITLE
mon: incorrect MAX AVAIL in "ceph df"

### PR DESCRIPTION
**Pull request by Liuchang0812**

### CODE REVIEW

**Comment by Liuchang0812** [comment]
L & K releases have problems in `ceph df`. 

master branch:

```
➜  build git:(wip) ✗ ceph osd erasure-code-profile set myprofile \
>    k=3 \
>    m=2 \
>    crush-failure-domain=osd
2017-09-05 21:01:09.564624 7f415a614700 -1 WARNING: all dangerous and experimental features are enabled.
2017-09-05 21:01:09.589226 7f415a614700 -1 WARNING: all dangerous and experimental features are enabled.
➜  build git:(wip) ✗ ceph osd pool create default2-ec-pool 12 12 erasure myprofile
2017-09-05 21:01:19.972730 7fe416634700 -1 WARNING: all dangerous and experimental features are enabled.
2017-09-05 21:01:20.005141 7fe416634700 -1 WARNING: all dangerous and experimental features are enabled.
pool 'default2-ec-pool' created
➜  build git:(wip) ✗ ./bin/ceph df                                                
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
2017-09-05 21:01:23.993915 7ffbe9313700 -1 WARNING: all dangerous and experimental features are enabled.
2017-09-05 21:01:24.001296 7ffbe9313700 -1 WARNING: all dangerous and experimental features are enabled.

GLOBAL:
    SIZE     AVAIL     RAW USED     %RAW USED 
    413G      115G         298G         72.23 
POOLS:
    NAME                  ID     USED      %USED     MAX AVAIL     OBJECTS 
    cephfs_data_a         1          0         0          111G           0 
    cephfs_metadata_a     2      2.19K         0          111G          21 
    isa-ec-pool           3          0         0          111G           0 
    default-ec-pool       4          0         0          111G           0 
    default2-ec-pool      5          0         0          111G           0 
```

kraken branch(11.2.0)

```
ceph df
2017-09-06 15:45:03.627312 7f550e4ac700 -1 WARNING: the following dangerous and experimental features are enabled: bluestore,rocksdb
2017-09-06 15:45:03.648622 7f550e4ac700 -1 WARNING: the following dangerous and experimental features are enabled: bluestore,rocksdb
GLOBAL:
    SIZE      AVAIL     RAW USED     %RAW USED 
    1559T     1034T         525T         33.68 
POOLS:
    NAME                                    ID     USED       %USED     MAX AVAIL     OBJECTS  
    rbd                                     0           0         0          193T            0 
    .rgw.root                               1        2937         0          193T            5 
    default.rgw.control                     2           0         0          193T            8 
    default.rgw.data.root                   3        4310         0          193T           14 
    default.rgw.gc                          4           0         0          193T           32 
    default.rgw.lc                          5           0         0          193T           32 
    default.rgw.log                         6           0         0          193T          128 
    default.rgw.users.uid                   7        1036         0          193T            6 
    default.rgw.users.keys                  8          31         0          193T            3 
    default.rgw.buckets.index               9           0         0          193T          112 
    default.rgw.buckets.data.deprecated     10       101G      0.05          193T        41431 
    default.rgw.buckets.non-ec              11          0         0          193T          385 
    default.rgw.buckets.data                12     14319G      3.12          434T     19418597 
```

**Comment by Liuchang0812** [comment]
@liewegas @tchaikov this patch fixes L release's df. take a look please.

File: qa/standalone/mon/osd-pool-df.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library Public License for more details.
+#
+source $CEPH_ROOT/qa/standalone/ceph-helpers.sh
+
+function run() {
+    local dir=$1
+    shift
+
+    export CEPH_MON="127.0.0.1:7105" # git grep '\<7105\>' : there must be only one
```
Comment (tchaikov) [inline qa/standalone/mon/osd-pool-df.sh]: please don't reuse port number. and i'd recommend updating `src/test/mon/PGMap.cc` to exercise this fix.

**Comment by tchaikov** [comment]
also, please fold the 2 commits into a single one.

File: qa/standalone/mon/osd-pool-df.sh (hunk 1 above)
Comment (Liuchang0812) [reply qa/standalone/mon/osd-pool-df.sh]: ooops....I forget to update port number. thank you

The logic is rather complicated, and existing unittest(replicated pool only) also misses `raw_used_rate` . so I add a standalone test. this standalone test is clear and straightforward.

adding some unittestes for EC pool to `src/test/mon/PGMap.cc` is necessary. I will do it.

**Comment by Liuchang0812** [comment]
changelog:

1. squash two commits
1. update MON_PORT to 7113
```
➜  qa git:(wip-max-avail-in-df) ✗ git grep '\<7112\>'
standalone/erasure-code/test-erasure-eio.sh:    export CEPH_MON="127.0.0.1:7112" # git grep '\<7112\>' : there must be only one
➜  qa git:(wip-max-avail-in-df) ✗ git grep '\<7113\>'
➜  qa git:(wip-max-avail-in-df) ✗ git grep '\<7114\>'

```

read `PGMap` unittest again. it seems that no need to add EC pool test case specialized. `raw_used_rate` param is [passed directly via `pool.size`](https://github.com/ceph/ceph/blob/bc27d9334822f164247782e65859a858783b19b1/src/test/mon/PGMap.cc#L207)

**Comment by Liuchang0812** [comment]
ping

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-09-07_18:45:31-rados-wip-yuri-testing3-2017-09-07-1719-distro-basic-smithi/

**Comment by Liuchang0812** [comment]
maybe we should back-port this patch to Luminous. A lot of people use this information in their dashboard.

**Comment by tchaikov** [comment]
i'd file a tracker ticket and label it with 'backport=luminous', etc. and wait for the backport team to pick it up, or just create backport PR targeting luminous.

**Comment by Liuchang0812** [comment]
http://tracker.ceph.com/issues/21243 is it ok?

**Comment by tchaikov** [comment]
no, you need to label it with 'backport=luminous' and mark it "pending backport", instead of moving it under the backport tracker.

**Comment by Liuchang0812** [comment]
:wave:  roger that. thank you ~

**Pull request merged**